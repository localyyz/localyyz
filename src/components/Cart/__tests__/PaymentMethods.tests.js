import React from "react"
import PaymentMethod from "../components/PaymentMethods"
import Renderer from "react-test-renderer"
import Colours from "../../../constants/colours"

const props = {
    cartStore:{
        paymentDetails: {},
        usePaymentMethod: () => {}
    },
    cartUiStore: {
        isPaymentVisible: true,
        togglePayment: () => {},
        getCheckoutSummary: () => {}
    },
    toggle: () => {}
}

describe("Payment Methods", () => {

    it("Payment Methods: should render properly", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).toJSON()
        expect(rendered).toMatchSnapshot()
    })

    it("Payment Methods: should be able to enter a valid credit card", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).getInstance().wrappedInstance
        rendered.onCardUpdate("4242424242424242")
        rendered.onBlur("card", !rendered.isCardValid("4242424242424242"))
        expect(rendered.state.number).toBe("4242 4242 4242 4242")
        expect(rendered.state.numberValid).toBe(true)
        expect(rendered.state.isCardInvalid).toBe(null)
        expect(rendered.refs.number.props.value).toBe("4242 4242 4242 4242")
        expect(rendered.refs.number.props.style[0].color).toBe(Colours.EmphasizedText)
    })

    it("Payment Methods: should not be able to enter an invalid credit card", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).getInstance().wrappedInstance
        rendered.onCardUpdate("123")
        rendered.onBlur("card", !rendered.isCardValid("123"))
        expect(rendered.state.number).toBe("123")
        expect(rendered.state.numberValid).toBe(false)
        expect(rendered.state.isCardInvalid).toBe(null)
        expect(rendered.refs.number.props.value).toBe("123") 
        expect(rendered.refs.number.props.style[1].color).toBe(Colours.Fail)
    })

    it("Payment Methods: should be able to enter a valid expiry", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).getInstance().wrappedInstance
        rendered.onExpiryUpdate("12/20")
        rendered.onBlur("expiry", !rendered.isExpiryValid("12/20"))
        expect(rendered.state.expiry).toBe("12/20")
        expect(rendered.state.expiryMonth).toBe("12")
        expect(rendered.state.expiryYear).toBe("20")
        expect(rendered.state.expiryValid).toBe(true)
        expect(rendered.refs.expiry.props.value).toBe("12/20")
        expect(rendered.refs.expiry.props.style[0].color).toBe(Colours.EmphasizedText)
    })

    it("Payment Methods: should not be able to enter an invalid expiry", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).getInstance().wrappedInstance
        rendered.onExpiryUpdate("00/00")
        rendered.onBlur("expiry", !rendered.isExpiryValid("00/00"))
        expect(rendered.state.expiry).toBe("00/00")
        expect(rendered.state.expiryMonth).toBe("00")
        expect(rendered.state.expiryYear).toBe("00")
        expect(rendered.state.expiryValid).toBe(false)
        expect(rendered.refs.expiry.props.value).toBe("00/00")
        expect(rendered.refs.expiry.props.style[1].color).toBe(Colours.Fail)
    })
    
    it("Payment Methods: should be able to enter a valid cvc", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).getInstance().wrappedInstance
        rendered.onCvcUpdate("123")
        rendered.onBlur("cvc", !rendered.isCvcValid("123"))
        expect(rendered.state.cvc).toBe("123")
        expect(rendered.state.cvcValid).toBe(true)
        expect(rendered.refs.cvc.props.value).toBe("123")
        expect(rendered.refs.cvc.props.style[0].color).toBe(Colours.EmphasizedText)
    })

    it("Payment Methods: should not be able to enter an invalid cvc", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).getInstance().wrappedInstance
        rendered.onCvcUpdate("12")
        rendered.onBlur("cvc", !rendered.isCvcValid("12"))
        expect(rendered.state.cvc).toBe("12")
        expect(rendered.state.cvcValid).toBe(false)
        expect(rendered.refs.cvc.props.value).toBe("12")
        expect(rendered.refs.cvc.props.style[1].color).toBe(Colours.Fail)
    })

    it("Payment Methods: should be able to a valid card holder name", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).getInstance().wrappedInstance
        rendered.onNameUpdate("Johnny Appleseed")
        rendered.onBlur("name", !rendered.isNameValid("Johnny Appleseed"))
        expect(rendered.state.name).toBe("Johnny Appleseed")
        expect(rendered.state.nameValid).toBe(true)
        expect(rendered.state.isNameInvalid).toBe(null)
        expect(rendered.refs.name.props.value).toBe("Johnny Appleseed")
        expect(rendered.refs.name.props.style[0].color).toBe(Colours.EmphasizedText)
    })

    it("Payment Methods: should be able to enter an invalid card holder name", () => {
        let rendered = Renderer.create(<PaymentMethod {...props}/>).getInstance().wrappedInstance
        rendered.onNameUpdate("Johnny")
        rendered.onBlur("name", !rendered.isNameValid("Johnny"))
        expect(rendered.state.name).toBe("Johnny")
        expect(rendered.state.nameValid).toBe(false)
        expect(rendered.state.isNameInvalid).toBe(null)
        expect(rendered.refs.name.props.value).toBe("Johnny")
        expect(rendered.refs.name.props.style[1].color).toBe(Colours.Fail)
    })

    it("Payment Methods: should be able to use a card if all entries are valid", () => {
        const propsLocal = {
            cartStore:{
                paymentDetails: {
                    number: "4242424242424242",
                    expiry: "12/20",
                    cvc: "230",
                    name: "Johnny Appleseed"
                },
                usePaymentMethod: () => {}
            },
            cartUiStore: {
                isPaymentVisible: true,
                togglePayment: () => {},
                getCheckoutSummary: () => {}
            },
        }
        let rendered = Renderer.create(<PaymentMethod {...propsLocal}/>).getInstance().wrappedInstance
        rendered.onSubmit()
        expect(rendered.refs.paymentIcon.props.icon.props.name).toBe("check-circle")
    })

    it("Payment Methods: should not be able to use a card if an entry is wrong", () => {
        const propsLocal = {
            cartStore:{
                paymentDetails: {
                    number: "4242424242424242",
                    expiry: "12/00",
                    cvc: "23",
                    name: "Johnny Appleseed"
                },
                usePaymentMethod: () => {}
            },
            cartUiStore: {
                isPaymentVisible: true,
                togglePayment: () => {},
                getCheckoutSummary: () => {}
            },
        }
        let rendered = Renderer.create(<PaymentMethod {...propsLocal}/>).getInstance().wrappedInstance
        rendered.onSubmit()
        expect(rendered.refs.paymentIcon.props.icon.props.name).toBe("dot-single")
    })

})