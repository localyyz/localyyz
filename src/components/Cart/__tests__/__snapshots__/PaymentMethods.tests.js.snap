// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Methods Payment Methods: should render properly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 15,
          "marginVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Payment
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginHorizontal": 7.5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "rgb(30, 30, 30)",
              "fontSize": 16,
              "fontWeight": "300",
            }
          }
        >
          no payment method selected
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#FF0000",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Entypo",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "paddingVertical": 7.5,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 2,
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingBottom": 3.75,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "opacity": 0,
                    "transform": Array [
                      Object {
                        "scale": 0.3,
                      },
                    ],
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": null,
                        "fontSize": 10,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "rgb(30, 30, 30)",
                    "flex": 1,
                    "fontSize": 12,
                    "fontWeight": "300",
                    "marginLeft": 3.75,
                  },
                  Object {
                    "color": null,
                  },
                ]
              }
            >
              CREDIT CARD
            </Text>
          </View>
          <View>
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              autoFocus={true}
              keyboardType="numeric"
              maxLength={19}
              onChangeText={[Function]}
              onEndEditing={[Function]}
              placeholder="0000 0000 0000 0000"
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "#000000",
                    "flex": 1,
                    "fontSize": 14,
                  },
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingBottom": 3.75,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "opacity": 0,
                    "transform": Array [
                      Object {
                        "scale": 0.3,
                      },
                    ],
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": null,
                        "fontSize": 10,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "rgb(30, 30, 30)",
                    "flex": 1,
                    "fontSize": 12,
                    "fontWeight": "300",
                    "marginLeft": 3.75,
                  },
                  Object {
                    "color": null,
                  },
                ]
              }
            >
              EXPIRY
            </Text>
          </View>
          <View>
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              keyboardType="numeric"
              maxLength={5}
              onChangeText={[Function]}
              onEndEditing={[Function]}
              placeholder="00/00"
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "#000000",
                    "flex": 1,
                    "fontSize": 14,
                  },
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingBottom": 3.75,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "opacity": 0,
                    "transform": Array [
                      Object {
                        "scale": 0.3,
                      },
                    ],
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": null,
                        "fontSize": 10,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "rgb(30, 30, 30)",
                    "flex": 1,
                    "fontSize": 12,
                    "fontWeight": "300",
                    "marginLeft": 3.75,
                  },
                  Object {
                    "color": null,
                  },
                ]
              }
            >
              CVC
            </Text>
          </View>
          <View>
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              keyboardType="numeric"
              maxLength={3}
              onChangeText={[Function]}
              onEndEditing={[Function]}
              placeholder="000"
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "#000000",
                    "flex": 1,
                    "fontSize": 14,
                  },
                  false,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingBottom": 3.75,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "opacity": 0,
                  "transform": Array [
                    Object {
                      "scale": 0.3,
                    },
                  ],
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": null,
                      "fontSize": 10,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "rgb(30, 30, 30)",
                  "flex": 1,
                  "fontSize": 12,
                  "fontWeight": "300",
                  "marginLeft": 3.75,
                },
                Object {
                  "color": null,
                },
              ]
            }
          >
            CARDHOLDER NAME
          </Text>
        </View>
        <View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            autoCorrect={false}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            placeholder="Johnny Appleseed"
            returnKeyType="next"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "flex": 1,
                  "fontSize": 14,
                },
                false,
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "marginHorizontal": 15,
            "marginVertical": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#9013FE",
                "borderRadius": 15,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 30,
                "paddingVertical": 7.5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "borderBottomColor": "#FFFFFF",
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "600",
                }
              }
            >
              USE THIS CARD
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 93.75,
          "paddingHorizontal": 15,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 14,
            "marginRight": 15,
          }
        }
      >
        ⚡️
      </Text>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#999999",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 12,
                "marginVertical": 1.5,
              },
              Object {
                "color": "#000000",
                "fontWeight": "600",
              },
            ]
          }
        >
          Payments powered by Stripe
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#999999",
              "flex": 1,
              "flexWrap": "wrap",
              "fontSize": 12,
              "marginVertical": 1.5,
            }
          }
        >
          Your credit card data is handled securely by Stripe and never stored on our servers
        </Text>
      </View>
    </View>
  </View>
</View>
`;
